mod/01_core/set/all/initialize.rb

```diff
@@ -1,13 +1,24 @@
-JUNK_INIT_ARGS = %w{ missing skip_virtual id }
+JUNK_INIT_ARGS = %w{ find_unused_name missing skip_virtual id }
 
 module ClassMethods
   def new args={}, options={}
     args = (args || {}).stringify_keys
+    args['name'] = find_unused_name args['name'] if args['find_unused_name']
     JUNK_INIT_ARGS.each { |a| args.delete(a) }
     %w{ type type_code }.each { |k| args.delete(k) if args[k].blank? }
     args.delete('content') if args['attach'] # should not be handled here!
     super args
   end
+
+  def find_unused_name base_name
+    test_name = base_name
+    add = 1
+    while Card.exists?(test_name) do
+      test_name = "#{base_name}#{add}"
+      add +=1
+    end
+    test_name
+  end
 end
```

Comment:
did you look at autoname?  It does pretty much this.  It might be worth improving that code to do this, but I'm not sure we need two separate methods.

In any event, we can't merge this as is because it reduces code coverage.  We don't have to unit test every bit of code, but we do need to make sure the code at least gets touched.